src/IBreadToken.sol

```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.22;
```

Comment:
this is important because 0.8.22 because "Solidity 0.8.22 introduces an overflow check optimization that automatically generates an unchecked arithmetic increment of the counter of for loops." https://soliditylang.org/blog/2023/10/25/solidity-0.8.22-release-announcement/